### DISCUSSION TITLE
CA-QC Notes

### DISCUSSION THREAD

**Discussion by pjakobsen** [electricitymaps/electricitymaps-contrib#3239] [General]
This is intended as a starting point to build, repair, or contribute in some other way to the completion of the map for Canada.  It's a reference manual for myself  as I learn to build and maintain the Quebec parser.   Perhaps someone else will find it useful.  

### Motivation 

Canada has one of the cleanest electricity grids in the world.  Canada constitutes  6.1% of planetary landmass, second only to Russia at 11%.  This translates into an abundance of moving  water that drive hydro electric dams.     

Some key facts from [Natural Resources Canada:](https://www.nrcan.gc.ca/science-and-data/data-and-analysis/energy-data-and-analysis/energy-facts/electricity-facts/20068)

- 67% of Canada’s electricity comes from renewable sources and 82% from non-GHG emitting sources
- Canada is the world’s third largest producer of hydroelectricity
- Canada exports about 8% of the electricity it generates to the United States. There are 34 active major international transmission lines connecting Canada to the U.S.

#### Quebec
https://github.com/tmrowco/electricitymap-contrib/issues/1521
https://github.com/tmrowco/electricitymap-contrib/issues/3218
https://github.com/tmrowco/electricitymap-contrib/pull/3237
https://github.com/tmrowco/electricitymap-contrib/pull/3254

### Conda vs. Pyenv

Both pyenv and conda both provide ways to specify the Python version to create new virtual environments.  This is a requirement if not running python 3.6 or 3.7

After `brew install miniconda` it  is quite simple to install and use multiple python versions with Conda. 

```
$python3 --version
Python 3.9.6
$conda create -n electric python=3.7
...
$ conda activate electric
$python --version
Python 3.7.11
```

Then run 
```
$brew install tesseract
$pip install poetry 
$poetry install -E parsers.    
$poetry run test 

```

### Testing Parser while working

To test a single parser while working on it:

https://github.com/tmrowco/electricitymap-contrib/wiki/Set-up-local-environment

### Regions

Regions in Canada can be found by linking to the raw zones.json file. (for example, using an interactive iPython terminal)
```
import requests
r = requests.get("https://raw.githubusercontent.com/tmrowco/electricitymap-contrib/master/config/zones.json")
zones = r.json()
canada = [zone for zone in zones if "CA-" in zone]
canada 

['CA-AB',
 'CA-BC',
 'CA-MB',
 'CA-NB',
 'CA-NL-NF',
 'CA-NL-LB',
 'CA-NS',
 'CA-NT',
 'CA-NU',
 'CA-ON',
 'CA-PE',
 'CA-QC',
 'CA-SK',
 'CA-YT']
```

### Contributors 
```
for c in canada:
    try:
        print(c,zones[c]['contributors'])
    except KeyError:
        print(c)

CA-AB ['https://github.com/mlucchini']
CA-BC ['https://github.com/daveachuk']
CA-MB
CA-NB ['https://github.com/jarek']
CA-NL-NF
CA-NL-LB
CA-NS ['https://github.com/jarek', 'https://github.com/corradio']
CA-NT
CA-NU
CA-ON ['https://github.com/corradio', 'https://github.com/jarek', 'https://github.com/scriptator', 'https://github.com/felixvanoost']
CA-PE ['https://github.com/reinvented', 'https://github.com/jarek']
CA-QC
CA-SK
CA-YT ['https://github.com/jarek', 'https://github.com/nessie2013']
```

Not all contributors are listed in the zones.json files.   To find out more, look for each individual parser and check commit history. For example, [Alberta - CA_AB](https://github.com/tmrowco/electricitymap-contrib/blob/master/parsers/CA_AB.py) 

## Province Notes

These notes are a summary from various discussions 

#### Quebec

Hydro-Québec says the "autres" category in its source (production.json) is "Sum of the estimated output from facilities belonging to independent power producers using different sources (biomass, biogas or hydropower), from whom Hydro‑Québec can purchase some or all of the output." Rather than putting this category in "unknown", I would put this in "biomass", since it's all renewable, and mostly biomass.

Québec's thermal generation is at Bécancour gas turbine. It is reported with a delay, and data source returning 0.0 can indicate either no generation or not-yet-reported generation. To handle this, if reported value is 0.0, overwrite it to None, so that backend can know this is not entirely reliable and might be updated later.

Set geothermal to 0 as there are no geothermal electricity generation stations in Québec (and all of Canada for that matter).

### Quick Intro to Canadian Electricity Market

More to come .....

### The Missing Parsers (and why).

More to come....

### Github  Tips

This is a large project with many collaborators.   There are features of Github used here that may be confusing for someone used to working on smaller projects .  Here are some quick tips:

- More to come....

**Comment by skovhus** [comment]
Thanks for creating this @pjakobsen!
